src/view/subgraph.rs

```diff
@@ -0,0 +1,347 @@
+//! Views into non-hierarchical parts of `PortView`s.
+
+use std::collections::{BTreeSet, HashMap};
+
+use crate::{algorithms::ConvexChecker, Direction, LinkView, NodeIndex, PortIndex, PortView};
+
+use super::filter::PortFiltered;
+
+type NodeCallback = fn(NodeIndex, &SubgraphContext) -> bool;
+type PortCallback = fn(PortIndex, &SubgraphContext) -> bool;
+
+/// View into a subgraph of a PortView.
+///
+/// The subgraph is given by boundary edges that define where one "enters" the
+/// subgraph (incoming boundary edge) and one "leaves" it (outgoing boundary
+/// edges).
+///
+/// A subgraph is well-defined if the incoming/outgoing boundary edges partition
```

Comment:
Pedantic nitpick; This allows the following case.

- Graph: $1 \to 2 \to 3$
- Incoming boundary: $\{(2,3)\}$
- Outgoing boundary: $\{(1,2)\}$

And the subgraph is then $\{1,3\}$.

Reply:
Yes it does. Any subgraph can be specified in this way (the example you give is precisely test case on `l.334`). Is that a problem?